examples/discovery-search-app/cypress/integration/collection_filter/spec.ts

```diff
@@ -0,0 +1,142 @@
+    cy.server();
+    cy.fixture('collections/collections.json').as('collectionsJSON');
+    cy.route('GET', '**/collections?version=2019-01-01', '@collectionsJSON').as('getCollections');
+    cy.fixture('component_settings/componentSettings.json').as('componentSettingsJSON');
+    cy.route('GET', '**/component_settings?version=2019-01-01', '@componentSettingsJSON').as(
+      'getComponentSettings'
+    );
+    cy.fixture('query/query.json').as('queryJSON');
+    cy.route('POST', '**/query?version=2019-01-01', '@queryJSON').as('postQuery');
+    cy.visit('/');
+    cy.wait(['@getCollections', '@getComponentSettings', '@postQuery']);
+
+    // Set up/override routes & fixtures that are specific to this file
+    cy.fixture('query/singleCollectionQuery.json').as('singleCollectionQueryJSON');
+    cy.fixture('query/doubleCollectionQuery.json').as('doubleCollectionQueryJSON');
+  });
+
+  describe('when the example app loads', () => {
+    it('the collection facet select does not appear', () => {
+      cy.get('.collection-facet-select').should('not.exist');
+    });
+  });
+
+  describe('when a query is made', () => {
+    beforeEach(() => {
+      cy.get('.bx--search-input').type('abil{enter}');
+      cy.wait('@postQuery');
+    });
+
+    it('the collection selector should appear', () => {
+      cy.get('#collection-facet-select').should('exist');
+      cy.get('#collection-facet-select')
+        .contains('Available collections')
+        .should('exist');
+    });
+
+    describe('and we click the collection filter', () => {
+      beforeEach(() => {
+        cy.get('#collection-facet-select').click();
+      });
+
+      it('displays a list of the collections in the project', () => {
+        cy.get('#collection-facet-select__menu')
+          .contains('deadspin')
+          .should('exist');
+        cy.get('#collection-facet-select__menu')
+          .contains('espn')
+          .should('exist');
+        cy.get('#collection-facet-select__menu')
+          .contains('finnegans wake')
+          .should('exist');
+      });
+
+      describe('and we select a single collection', () => {
+        beforeEach(() => {
+          cy.route('POST', '**/query?version=2019-01-01', '@singleCollectionQueryJSON').as(
+            'postQuerySingleCollection'
+          );
+          cy.get('.bx--list-box__menu-item')
+            .contains('finnegans wake')
+            .click();
+          cy.wait('@postQuerySingleCollection').as('singleCollectionQueryObject');
+        });
+
+        it('should make a query against only the selected collection', () => {
+          cy.get('@singleCollectionQueryObject')
+            //@ts-ignore
+            .its('requestBody.collection_ids')
+            .should('contain', 'paris19221939');
+        });
+
+        it('the clear all selected collections button appears', () => {
+          cy.get('div[aria-label="Clear Selection"]').should('exist');
```

Comment:
I wonder if data test id's would be better here. I'm envisioning a scenario where we get a second round of design revisions and we need to make tweaks to the naming we've used for certain buttons and labels